### PR TITLE
[WIP]: Simplify superset permissions

### PR DESCRIPTION
**Pull request by gaurav1999**
Coming from #7501 .

### CODE REVIEW

**Comment by gaurav1999** [comment]
@mistercrunch , I wish to finish the TODO's in #7501 , also will remove my author name from commit, as this is your work. If you say I can edit this MR for your branch.

**Comment by gaurav1999** [comment]
Right now, I can see the permissions which I added/edited but problem is, the older permissions still exists, they are not removed for some reason :-( . 

For example:
Screenshot from 2020-04-03 04-10-01
Screenshot from 2020-04-03 04-09-37

File: superset/views/schedules.py [hunk 1]
```diff
@@ -47,6 +47,9 @@ class EmailScheduleView(
     SupersetModelView, DeleteMixin
 ):  # pylint: disable=too-many-ancestors
     include_route_methods = RouteMethod.CRUD_SET
+
+    class_permission_name = "ScheduleEmail"
```
Comment (gaurav1999) [inline superset/views/schedules.py:51]: I am unable to see this view in the Permission set.

Screenshot from 2020-04-03 04-28-12

File: superset/views/chart/views.py [hunk 2]
```diff
@@ -33,6 +33,7 @@ class SliceModelView(
     SliceMixin, SupersetModelView, DeleteMixin
 ):  # pylint: disable=too-many-ancestors
     route_base = "/chart"
+    class_permission_name = "Chart"
```
Comment (gaurav1999) [inline superset/views/chart/views.py:36]: This comes out okay, but the previous permission still persists.
Screenshot from 2020-04-03 04-29-21
Screenshot from 2020-04-03 04-29-49

File: superset/views/annotations.py [hunk 3]
```diff
@@ -46,6 +46,9 @@ class AnnotationModelView(
     SupersetModelView, DeleteMixin
 ):  # pylint: disable=too-many-ancestors
     datamodel = SQLAInterface(Annotation)
+
+    class_permission_name = "Annotation"
```
Comment (gaurav1999) [inline superset/views/annotations.py:50]: Not sure about this Output either

File: superset/connectors/sqla/views.py [hunk 4]
```diff
@@ -167,6 +170,9 @@ class TableColumnInlineView(CompactCRUDMixin, SupersetModelView):
 
 class SqlMetricInlineView(CompactCRUDMixin, SupersetModelView):
     datamodel = SQLAInterface(models.SqlMetric)
+
+    class_permission_name = "Datasource"
```
Comment (gaurav1999) [inline superset/connectors/sqla/views.py:174]: Same as below, I have "Datasource" permission visible, but the older one still persists.

File: superset/connectors/druid/views.py [hunk 5]
```diff
@@ -190,6 +196,9 @@ class DruidMetricInlineView(CompactCRUDMixin, SupersetModelView):
 
 class DruidClusterModelView(SupersetModelView, DeleteMixin, YamlExportMixin):
     datamodel = SQLAInterface(models.DruidCluster)
+
+    class_permission_name = "DataSource"
```
Comment (mistercrunch) [inline superset/connectors/druid/views.py:200]: case is different

**Comment by mistercrunch** [comment]
@dpgaspar

File: superset/connectors/druid/views.py (hunk 5 above)
Comment (gaurav1999) [reply superset/connectors/druid/views.py:200]: sure, let me redo it

File: superset/views/core.py [hunk 6]
```diff
@@ -998,7 +998,7 @@ def save_or_overwrite_slice(
             )
         elif request.args.get("add_to_dash") == "new":
             # check create dashboard permissions
-            dash_add_perm = security_manager.can_access("can_add", "DashboardModelView")
+            dash_add_perm = security_manager.can_access("can_write", "DashboardModelView")
```
Comment (gaurav1999) [inline superset/views/core.py:1001]: I think I missed this one

File: superset/views/database/api.py [hunk 7]
```diff
@@ -113,12 +114,20 @@ class DatabaseRestApi(DatabaseMixin, BaseSupersetModelRestApi):
     datamodel = SQLAInterface(Database)
 
     include_route_methods = {"get_list", "table_metadata", "select_star"}
-    class_permission_name = "DatabaseView"
-    method_permission_name = {
+
```
Comment (gaurav1999) [inline superset/views/database/api.py:117]: @mistercrunch Had a doubt, Was wondering if this should be bundled in with the permission "Database"

**Review by gaurav1999 — COMMENTED** [review]
```
Computed security converge:
Add to Roles:
Where ('ChartRestApi', 'can_post') add {('SliceModelView', 'can_add')}
Where ('ChartRestApi', 'can_get') add {('SliceModelView', 'can_list'), ('SliceModelView', 'can_show')}
Where ('ChartRestApi', 'can_related') add {('SliceModelView', 'can_list')}
Where ('ChartRestApi', 'can_info') add {('SliceModelView', 'can_list')}
Where ('ChartRestApi', 'can_bulk_delete') add {('SliceModelView', 'can_delete')}
Where ('ChartRestApi', 'can_delete') add {('SliceModelView', 'can_delete')}
Where ('ChartRestApi', 'can_put') add {('SliceModelView', 'can_edit')}
Where ('DashboardRestApi', 'can_post') add {('DashboardModelView', 'can_add')}
Where ('DashboardRestApi', 'can_get') add {('DashboardModelView', 'can_show'), ('DashboardModelView', 'can_list')}
Where ('DashboardRestApi', 'can_related') add {('DashboardModelView', 'can_list')}
Where ('DashboardRestApi', 'can_info') add {('DashboardModelView', 'can_list')}
Where ('DashboardRestApi', 'can_bulk_delete') add {('DashboardModelView', 'can_delete')}
Where ('DashboardRestApi', 'can_delete') add {('DashboardModelView', 'can_delete')}
Where ('DashboardRestApi', 'can_put') add {('DashboardModelView', 'can_edit')}
Where ('DashboardRestApi', 'can_export') add {('DashboardModelView', 'can_mulexport')}
Where ('DatabaseRestApi', 'can_select_star') add {('DatabaseRestApi', 'can_list')}
Where ('DatabaseRestApi', 'can_table_metadata') add {('DatabaseRestApi', 'can_list')}
Where ('DatabaseRestApi', 'can_get') add {('DatabaseRestApi', 'can_list')}
Where ('DatasetRestApi', 'can_post') add {('TableModelView', 'can_add')}
Where ('DatasetRestApi', 'can_get') add {('TableModelView', 'can_list'), ('TableModelView', 'can_show')}
Where ('DatasetRestApi', 'can_related') add {('TableModelView', 'can_list')}
Where ('DatasetRestApi', 'can_info') add {('TableModelView', 'can_list')}
Where ('DatasetRestApi', 'can_delete') add {('TableModelView', 'can_delete')}
Where ('DatasetRestApi', 'can_put') add {('TableModelView', 'can_edit')}
Where ('DatasetRestApi', 'can_refresh') add {('TableModelView', 'can_edit')}
Where ('DatasetRestApi', 'can_export') add {('TableModelView', 'can_mulexport')}
Where ('QueryRestApi', 'can_get') add {('QueryView', 'can_show'), ('QueryView', 'can_list')}
Where ('AnnotationLayerModelView', 'can_show') add {('Annotation', 'can_read')}
Where ('AnnotationLayerModelView', 'can_list') add {('Annotation', 'can_read')}
Where ('AnnotationLayerModelView', 'can_add') add {('Annotation', 'can_write')}
Where ('AnnotationLayerModelView', 'muldelete') add {('Annotation', 'can_write')}
Where ('AnnotationLayerModelView', 'can_delete') add {('Annotation', 'can_write')}
Where ('AnnotationLayerModelView', 'can_edit') add {('Annotation', 'can_write')}
Where ('AnnotationModelView', 'can_show') add {('Annotation', 'can_read')}
Where ('AnnotationModelView', 'can_list') add {('Annotation', 'can_read')}
Where ('AnnotationModelView', 'can_add') add {('Annotation', 'can_write')}
Where ('AnnotationModelView', 'muldelete') add {('Annotation', 'can_write')}
Where ('AnnotationModelView', 'can_delete') add {('Annotation', 'can_write')}
Where ('AnnotationModelView', 'can_edit') add {('Annotation', 'can_write')}
Where ('DatabaseView', 'yaml_export') add {('DatabaseView', 'can_read')}
Where ('DatabaseView', 'can_show') add {('DatabaseView', 'can_read')}
Where ('DatabaseView', 'can_list') add {('DatabaseView', 'can_read')}
Where ('DatabaseView', 'can_add') add {('DatabaseView', 'can_write')}
Where ('DatabaseView', 'muldelete') add {('DatabaseView', 'can_write')}
Where ('DatabaseView', 'can_delete') add {('DatabaseView', 'can_write')}
Where ('DatabaseView', 'can_edit') add {('DatabaseView', 'can_write')}
Where ('SliceModelView', 'can_show') add {('Chart', 'can_read')}
Where ('SliceModelView', 'can_list') add {('Chart', 'can_read')}
Where ('SliceModelView', 'can_add') add {('Chart', 'can_write')}
Where ('SliceModelView', 'can_delete') add {('Chart', 'can_write')}
Where ('SliceModelView', 'can_edit') add {('Chart', 'can_write')}
Where ('SliceModelView', 'muldelete') add {('Chart', 'can_write')}
Where ('SliceModelView', 'can_download') add {('Chart', 'can_write')}
Where ('DashboardModelView', 'mulexport') add {('Dashboard', 'can_read')}
Where ('DashboardModelView', 'can_download_dashboards') add {('Dashboard', 'can_read')}
Where ('DashboardModelView', 'can_show') add {('Dashboard', 'can_read')}
Where ('DashboardModelView', 'can_list') add {('Dashboard', 'can_read')}
Where ('DashboardModelView', 'can_add') add {('Dashboard', 'can_write')}
Where ('DashboardModelView', 'muldelete') add {('Dashboard', 'can_write')}
Where ('DashboardModelView', 'can_delete') add {('Dashboard', 'can_write')}
Where ('DashboardModelView', 'can_edit') add {('Dashboard', 'can_write')}
Where ('CssTemplateModelView', 'can_show') add {('CssTemplate', 'can_read')}
Where ('CssTemplateModelView', 'can_list') add {('CssTemplate', 'can_read')}
Where ('CssTemplateModelView', 'can_add') add {('CssTemplate', 'can_write')}
Where ('CssTemplateModelView', 'muldelete') add {('CssTemplate', 'can_write')}
Where ('CssTemplateModelView', 'can_delete') add {('CssTemplate', 'can_write')}
Where ('CssTemplateModelView', 'can_edit') add {('CssTemplate', 'can_write')}
Where ('QueryView', 'can_show') add {('Query', 'can_read')}
Where ('QueryView', 'can_list') add {('Query', 'can_read')}
Where ('CssTemplateAsyncModelView', 'can_list') add {('CssTemplate', 'can_read')}
Where ('CssTemplateAsyncModelView', 'muldelete') add {('CssTemplate', 'can_write')}
Where ('DashboardModelViewAsync', 'mulexport') add {('Dashboard', 'can_read')}
Where ('DashboardModelViewAsync', 'can_list') add {('Dashboard', 'can_read')}
Where ('DashboardModelViewAsync', 'muldelete') add {('Dashboard', 'can_write')}
Where ('SavedQueryView', 'can_show') add {('SavedQuery', 'can_read')}
Where ('SavedQueryView', 'can_list') add {('SavedQuery', 'can_read')}
Where ('SavedQueryView', 'can_add') add {('SavedQuery', 'can_write')}
Where ('SavedQueryView', 'muldelete') add {('SavedQuery', 'can_write')}
Where ('SavedQueryView', 'can_delete') add {('SavedQuery', 'can_write')}
Where ('SavedQueryView', 'can_edit') add {('SavedQuery', 'can_write')}
Where ('SavedQueryViewApi', 'can_show') add {('SavedQuery', 'can_read')}
Where ('SavedQueryViewApi', 'can_list') add {('SavedQuery', 'can_read')}
Where ('SavedQueryViewApi', 'can_add') add {('SavedQuery', 'can_write')}
Where ('SavedQueryViewApi', 'muldelete') add {('SavedQuery', 'can_write')}
Where ('SavedQueryViewApi', 'can_edit') add {('SavedQuery', 'can_write')}
Where ('SliceAsync', 'can_list') add {('Chart', 'can_read')}
Where ('SliceAsync', 'muldelete') add {('Chart', 'can_write')}
Where ('SqlMetricInlineView', 'can_show') add {('Datasource', 'can_read')}
Where ('SqlMetricInlineView', 'can_list') add {('Datasource', 'can_read')}
Where ('SqlMetricInlineView', 'can_add') add {('Datasource', 'can_write')}
Where ('SqlMetricInlineView', 'can_delete') add {('Datasource', 'can_write')}
Where ('SqlMetricInlineView', 'can_edit') add {('Datasource', 'can_write')}
Where ('TableColumnInlineView', 'can_show') add {('Datasource', 'can_read')}
Where ('TableColumnInlineView', 'can_list') add {('Datasource', 'can_read')}
Where ('TableColumnInlineView', 'can_add') add {('Datasource', 'can_write')}
Where ('TableColumnInlineView', 'can_delete') add {('Datasource', 'can_write')}
Where ('TableColumnInlineView', 'can_edit') add {('Datasource', 'can_write')}
Where ('TableModelView', 'yaml_export') add {('Datasource', 'can_read')}
Where ('TableModelView', 'can_show') add {('Datasource', 'can_read')}
Where ('TableModelView', 'can_list') add {('Datasource', 'can_read')}
Where ('TableModelView', 'can_add') add {('Datasource', 'can_write')}
Where ('TableModelView', 'muldelete') add {('Datasource', 'can_write')}
Where ('TableModelView', 'can_delete') add {('Datasource', 'can_write')}
Where ('TableModelView', 'can_edit') add {('Datasource', 'can_write')}
Where ('TableModelView', 'refresh') add {('Datasource', 'refresh')}
Where ('LogRestApi', 'can_get') add {('Log', 'can_show'), ('Log', 'can_list')}
Where ('LogRestApi', 'can_post') add {('Log', 'can_post')}
Where ('LogModelView', 'can_show') add {('Log', 'can_read')}
Where ('LogModelView', 'can_list') add {('Log', 'can_read')}
Del from Roles:
('AnnotationModelView', 'can_edit')
('QueryRestApi', 'can_get')
('DatasetRestApi', 'can_delete')
('DatasetRestApi', 'can_post')
('SavedQueryViewApi', 'can_list')
('TableModelView', 'muldelete')
('ChartRestApi', 'can_info')
('SqlMetricInlineView', 'can_edit')
('ChartRestApi', 'can_post')
('SliceModelView', 'muldelete')
('CssTemplateAsyncModelView', 'can_list')
('ChartRestApi', 'can_put')
('DatabaseView', 'can_add')
('LogModelView', 'can_list')
('DatasetRestApi', 'can_related')
('TableColumnInlineView', 'can_show')
('DatasetRestApi', 'can_put')
('AnnotationLayerModelView', 'can_show')
('CssTemplateModelView', 'can_edit')
('DatabaseView', 'can_delete')
('SavedQueryView', 'can_edit')
('DashboardModelViewAsync', 'can_list')
('SavedQueryView', 'can_add')
('DashboardModelViewAsync', 'mulexport')
('DashboardRestApi', 'can_export')
('TableColumnInlineView', 'can_edit')
('DatabaseView', 'can_edit')
('SavedQueryView', 'can_delete')
('SavedQueryView', 'muldelete')
('LogRestApi', 'can_get')
('DashboardRestApi', 'can_post')
('SavedQueryViewApi', 'can_add')
('DashboardRestApi', 'can_info')
('LogModelView', 'can_show')
('TableColumnInlineView', 'can_list')
('DatabaseRestApi', 'can_table_metadata')
('DatabaseRestApi', 'can_select_star')
('SliceAsync', 'muldelete')
('DashboardRestApi', 'can_delete')
('SqlMetricInlineView', 'can_add')
('DatabaseView', 'can_show')
('AnnotationLayerModelView', 'can_delete')
('AnnotationModelView', 'can_show')
('CssTemplateModelView', 'can_list')
('SavedQueryViewApi', 'muldelete')
('TableColumnInlineView', 'can_add')
('DatasetRestApi', 'can_refresh')
('AnnotationLayerModelView', 'can_add')
('SqlMetricInlineView', 'can_delete')
('SqlMetricInlineView', 'can_list')
('DatasetRestApi', 'can_get')
('SavedQueryViewApi', 'can_edit')
('DashboardRestApi', 'can_bulk_delete')
('CssTemplateModelView', 'muldelete')
('CssTemplateModelView', 'can_delete')
('DatasetRestApi', 'can_export')
('ChartRestApi', 'can_get')
('SliceAsync', 'can_list')
('CssTemplateModelView', 'can_add')
('SavedQueryView', 'can_list')
('DashboardModelView', 'muldelete')
('DatabaseView', 'can_list')
('AnnotationModelView', 'can_list')
('DashboardModelViewAsync', 'muldelete')
('AnnotationModelView', 'can_delete')
('DatasetRestApi', 'can_info')
('SavedQueryViewApi', 'can_show')
('AnnotationLayerModelView', 'can_edit')
('SqlMetricInlineView', 'can_show')
('AnnotationModelView', 'can_add')
('DashboardModelView', 'can_download_dashboards')
('ChartRestApi', 'can_related')
('CssTemplateAsyncModelView', 'muldelete')
('DashboardModelView', 'mulexport')
('TableModelView', 'refresh')
('DatabaseView', 'yaml_export')
('CssTemplateModelView', 'can_show')
('LogRestApi', 'can_post')
('DatabaseRestApi', 'can_get')
('SliceModelView', 'can_download')
('ChartRestApi', 'can_bulk_delete')
('DashboardRestApi', 'can_get')
('TableColumnInlineView', 'can_delete')
('DashboardRestApi', 'can_related')
('DashboardRestApi', 'can_put')
('TableModelView', 'yaml_export')
('AnnotationLayerModelView', 'muldelete')
('AnnotationModelView', 'muldelete')
('DatabaseView', 'muldelete')
('ChartRestApi', 'can_delete')
('SavedQueryView', 'can_show')
('AnnotationLayerModelView', 'can_list')
Remove views:
DashboardModelViewAsync
ChartRestApi
LogRestApi
CssTemplateAsyncModelView
SavedQueryView
SavedQueryViewApi
AnnotationModelView
SliceAsync
TableColumnInlineView
AnnotationLayerModelView
CssTemplateModelView
SqlMetricInlineView
DatasetRestApi
DashboardRestApi
QueryRestApi
LogModelView
Remove permissions:
can_related
can_table_metadata
can_bulk_delete
can_refresh
mulexport
yaml_export
can_info
can_download
can_export
muldelete
can_download_dashboards
```

--dry-run output when I run it on a fresh db.

**Comment by amitmiran137** [comment]
closing due to inactivity

**Pull request closed without merging**